### PR TITLE
Rewrite the search run doc

### PR DESCRIPTION
**Pull request by chenmoneygithub**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/chenmoneygithub/mlflow/pull/10863?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/10863/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 10863
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

Rewrite the search run doc, the gist is:
- Document the syntax better.
- Provide a script to generate sample runs to test the query.
- Provide a better reading flow.

### How is this PR tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [x] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: docs/source/search-runs.rst [hunk 1]
```diff
@@ -3,207 +3,238 @@
...
+Briefly a search condition consists of 3 parts:
 
-.. contents:: In this section:
-  :local:
-  :depth: 1
+- An MLflow field on the left side, which can be some metric, param, tag, dataset or MLflow run metadata, e.g.,
+  ``metric.loss`` or ``param.batch_size``.
+- A comparator in the middle, we support 3 types of comparators:
+  - Numeric comparators: ``=``, ``!=``, ``>``, ``>=``, ``<``, and ``<=``.
+  - String comparators: ``=``, ``!=``, ``LIKE`` and ``ILIKE``.
```
Comment (WeichenXu123) [inline docs/source/search-runs.rst:125]: We should document whether `=` for string is case-sensitive or case-insensitive.

File: docs/source/search-runs.rst [hunk 2]
```diff
@@ -3,207 +3,238 @@
...
+  - Set comparators: ``IN``.
+- A reference value on the right side.
 
-Entity Names Containing Special Characters
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+.. note::
+  There is a caveat that using ``IN`` requires wrapping string with single quotes, e.g.,
+  ``params."learning rate" IN ('0.001', '0.01')``. ``params."learning_rate" IN ("0.001", "0.01")`` will throw an error.
```
Comment (WeichenXu123) [inline docs/source/search-runs.rst:131]: I remember we should use format `params.`learning rate`` ?

Comment (WeichenXu123) [reply docs/source/search-runs.rst:131]: > that using ``IN`` requires wrapping string with single quotes

Btw why we have this limitation ? I forgot related code details. Is it limitation in `sqlalchemy` parser ?

File: docs/source/search-runs.rst (hunk 1 above)
Comment (chenmoneygithub) [reply docs/source/search-runs.rst:125]: great call!

File: docs/source/search-runs.rst (hunk 2 above)
Comment (chenmoneygithub) [reply docs/source/search-runs.rst:131]: The weird fact is all the following work:
- params.`learning rate`
- params.'learning rate'
- params."learning rate"

I am only mentioning one way in the doc to reduce the users' confusion. 

> Btw why we have this limitation?

I don't know actually... just found this while playing with my code examples and mlflow UI. This is surely a bug to me.

File: docs/source/search-runs.rst [hunk 3]
```diff
@@ -3,207 +3,240 @@
...
 
-.. contents:: In this section:
-  :local:
-  :depth: 1
+- An MLflow field on the left side, which can be some metric, param, tag, dataset or MLflow run metadata, e.g.,
+  ``metric.loss`` or ``param.batch_size``.
+- A comparator in the middle, we support 3 types of comparators:
+  - Numeric comparators: ``=``, ``!=``, ``>``, ``>=``, ``<``, and ``<=``.
+  - String comparators: ``=``, ``!=``, ``LIKE`` and ``ILIKE``. String comparators are case sensitive.
+  - Set comparators: ``IN``.
```
Comment (WeichenXu123) [inline docs/source/search-runs.rst:126]: Let's also mention whether ``IN`` for string type is case-sensitive or case-insensitive. This is often confusing to user :)

File: docs/source/search-runs.rst [hunk 4]
```diff
@@ -3,207 +3,240 @@
...
+Briefly a search condition consists of 3 parts:
 
-.. contents:: In this section:
-  :local:
-  :depth: 1
+- An MLflow field on the left side, which can be some metric, param, tag, dataset or MLflow run metadata, e.g.,
+  ``metric.loss`` or ``param.batch_size``.
+- A comparator in the middle, we support 3 types of comparators:
+  - Numeric comparators: ``=``, ``!=``, ``>``, ``>=``, ``<``, and ``<=``.
+  - String comparators: ``=``, ``!=``, ``LIKE`` and ``ILIKE``. String comparators are case sensitive.
```
Comment (WeichenXu123) [inline docs/source/search-runs.rst:125]: > String comparators are case sensitive.

This phrase should be for ``==`` and ``!=`` and ``like`` comparator, ``ilike`` is case-insensitive.

File: docs/source/search-runs.rst (hunk 3 above)
Comment (chenmoneygithub) [reply docs/source/search-runs.rst]: good call! 

Actually I did more testing, and `IN` only works for dataset information and run metadata search. I've updated the guide with the correpsonding information.

File: docs/source/search-runs.rst (hunk 4 above)
Comment (chenmoneygithub) [reply docs/source/search-runs.rst]: thanks, modified

**Review by WeichenXu123 — APPROVED** [review]
LGTM :)
